features/networking/pod.feature

```diff
@@ -452,3 +452,56 @@ Feature: Pod related networking scenarios
     """
     OVN is functional on the cluster
     """
+  # @author anusaxen@redhat.com
+  # @case_id OCP-33413
+  @admin
+  @destructive
+  Scenario: xt_u32 kernel module functionality check from NET_ADMIN pods
+    Given I have a project
+    And I have a pod-for-ping in the project
+    Then evaluation of `pod.name` is stored in the :client_pod clipboard
+    Given I select a random node's host
+    #Making sure the iptables on latest centos are actually nf_tables as the xt_u32 is only supported with that
+    When I run commands on the host:
+      | iptables --version |
+    Then the step should succeed
+    Then the output should contain "nf_tables"
+
+    Given I obtain test data file "networking/centos_latest_admin.yaml"
+    When I run oc create as admin over "centos_latest_admin.yaml" replacing paths:
+      | ["metadata"]["name"]      | server-pod          |
+      | ["metadata"]["namespace"] | <%= project.name %> |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | name=centos-pod |
+    Then evaluation of `pod` is stored in the :server_pod clipboard
+    Given I use the "<%= cb.server_pod.node_name %>" node
+    #Enabling XT_u32 module on server pod's node
+    And I run commands on the host:
+      | modprobe xt_u32 |
+    Then the step should succeed
+
+    #Checking whether server pod also has nf_tables version installed
+    When admin executes on the "<%= cb.server_pod.name %>" pod:
+      | bash | -c | iptables --version |
+    Then the step should succeed
+    And the output should contain "nf_tables"
+
+    #Creating iptable rule on server pod with module u32 and Successful pings will represent a successful xt_u32 match
```

Comment:
@anuragthehatter Could you describe what's the detail rule the line 492 try to match? I mean you may do the same thing as the line 498 mentioning "less than 256 bytes".

Reply:
Sure its bit complicated actually. We are using xt_u32 module for creating bit level rules in an IP header. We are using ICMP/ping here. Maximum payload a ping packet can take is 56 bytes(excluding 8 header) so a 2000 bytes source will trigger some fragmentation. Detailed explanation for this rule is [here](http://www.stearns.org/doc/iptables-u32.current.html) and search ```Moving on to the TCP header``` section . 256 bytes test was taken from here as well and explains how iptables entry was designed for that.

In short we are extracting 6&0xFF=0x6 from the packet which will tell whether its a TCP packet. This is stored in byte 9 of the IP header, so we'll pull 4 bytes starting at byte 6, drop 6-8, and check to see if it's 6 and so on.

The first rule is actually just to check if -m 32 (module xt_u32) works/matches with iptables and 2nd rule is to check extended functionality like byte level ACCEPT/REJECT. Combining both rules may not be a good idea for debugging. At least if test fails at first rule, we know something is wrong with module and if fails at second, it might depict packet bit matching level issues.